applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_runge_kutta_4.h

```diff
@@ -0,0 +1,834 @@
+     * @param rModelPart The model part to be computed
+     * @param MoveMeshFlag The flag to set if the mesh is moved or not
+     */
+    explicit CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4(
+        ModelPart &rModelPart,
+        bool MoveMeshFlag = false,
+        int RebuildLevel = 0)
+        : BaseType(rModelPart, MoveMeshFlag, RebuildLevel)
+    {
+    }
+
+    /** Copy constructor.
+     */
+    CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4(const CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4 &Other) = delete;
+
+    /** Destructor.
+     */
+    virtual ~CompressibleNavierStokesExplicitSolvingStrategyRungeKutta4() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     * @return The default parameters
+     */
+    Parameters GetDefaultParameters() const override
+    {
+        Parameters default_parameters = Parameters(R"(
+        {
+            "name" : "compressible_navier_stokes_explicit_explicit_solving_strategy_runge_kutta_4",
+            "rebuild_level" : 0,
+            "move_mesh_flag": false,
+            "shock_capturing" : true
+        })");
+
+        // Getting base class default parameters
+        const Parameters base_default_parameters = BaseType::GetDefaultParameters();
+        default_parameters.RecursivelyAddMissingParameters(base_default_parameters);
+        return default_parameters;
+    }
+
+    /**
+     * @brief Returns the name of the class as used in the settings (snake_case format)
+     * @return The name of the class
+     */
+    static std::string Name()
+    {
+        return "compressible_navier_stokes_explicit_solving_strategy_runge_kutta_4";
+    }
+
+    /**
+     * @brief This method assigns settings to member variables
+     * @param ThisParameters Parameters that are assigned to the member variables
+     */
+    void AssignSettings(const Parameters ThisParameters) override
+    {
+        // Base class assign settings call
+        BaseType::AssignSettings(ThisParameters);
+
+        // Set the specific compressible NS settings
+        mShockCapturing = ThisParameters["shock_capturing"].GetBool();
+    }
+
+    /**
+     * @brief Initialization of member variables and prior operations
+     * In this method we call the base strategy initialize and initialize the time derivatives
+     * This is required to prevent OpenMP errors as the time derivatives are stored in the non-historical database
+     */
+    void Initialize() override
+    {
+        auto& r_model_part = BaseType::GetModelPart();
+        const auto& r_process_info = r_model_part.GetProcessInfo();
+
+        // Call the base RK4 finalize substep method
+        BaseType::Initialize();
+
+        // If required, initialize the OSS projection variables
+        if (r_process_info[OSS_SWITCH]) {
```

Comment:
it may make sense to have a function "ActivateOss" which a user could eventually activate at some point of the analysis

Reply:
You mean activate the OSS after the simulation has already started with ASGS? I guess the motivation would be to stabilize "less" after the initial transition has been completed. Is this your motivation?

Note that this would imply changes in the analysis stage or creating an OSS switch on/off process. Taking this into account, I'd leave for a future PR.